File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False;
        contador = [0]*26
        for i in range(len(s)):
            contador[ord(s[i]) - ord("a")]+=1
            contador[ord(t[i]) - ord("a")]-=1
        for i in contador:
            if i!=0: return False;
        return True;

File: Data Structures & Algorithms/string-encode-and-decode/submission-2.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string sol="";
        int contador=0;
        for (string s:strs){
            sol+=to_string(s.size())+'#'+s;

        }
        return sol;
    }

    vector<string> decode(string s) {
        vector<string> sol;
        int i=0;
        while(i<s.size())
        {
            int j=i;
            while(s[j]!='#'){
                j++;
            }
            int longitud = stoi( s.substr(i,j-i) );
            i=j+1;
            j=i+longitud;
            sol.push_back( s.substr(i, longitud) );
            i=j;
        }
        return sol;
    }
};
